lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -909,6 +985,45 @@ static NodeValue simplifyConcatNode(Function *F, ConcatNode *CN) {
     }
   }
 
+  /// If the inputs of concat are all reshape nodes, and the inputs of reshape
+  /// are able to be concatenated, then we can perform concat first and then
+  /// reshape. Concat(Reshape(x) * N) -> Reshape(Concat(x * N)).
+  {
+    std::vector<ReshapeNode *> reshapes;
+    bool optimizable = true;
+    for (auto &I : CN->getInputs()) {
+      if (auto *S = dyn_cast<ReshapeNode>(I)) {
+        // The input of Reshape nodes should have the same shape.
+        if (reshapes.size() == 0 ||
+            reshapes.front()->getInput().dims().size() ==
+                S->getInput().dims().size()) {
+          reshapes.push_back(S);
+          continue;
+        }
+      }
+      optimizable = false;
```

Comment:
if not optimizable we can return from here and remove redundant "optimizable" var completely.

Reply:
removed it